acond/src/restful.rs

```diff
@@ -492,13 +620,33 @@ impl ExchangeService {
 
         acond_io::read_async_lock(self.stream.clone()).await
     }
+
+    async fn refresh_hmac_key(&self) -> Result<PKey<Private>> {
+        let mut key = [0; 128];
+        rand::rand_bytes(&mut key)?;
+        let pkey = PKey::hmac(&key)?;
+
+        let mut hmac_key = self.hmac_key.write().await;
+        *hmac_key = Some(pkey.clone());
+
+        Ok(pkey)
+    }
+
+    async fn reset_hmac_key(&self) {
+        let mut hmac_key = self.hmac_key.write().await;
+        *hmac_key = None;
```

Comment:
Secret is not wiped. Suggest to refresh - i.e., a new rand will always overwrite the old one.

Reply:
`aconcli run failed`
```
[sgx@sgx-emr02 test]$ ATD_QEMU=qemu-img ATD_BIOS=OVMF.fd ATD_KERNEL=kernel.img ATD_RD=initrd.img ./aconcli run -n "$docker_id.json" -c :5532 -f "../scripts/acon-startvm"
2024/06/06 15:46:44 Created Virtual Machine, PID: 3101188, Connection: :5532
2024/06/06 15:46:44 Service: Connecting :5532
2024/06/06 15:46:44 no login required
map[Content-Type:[multipart/form-data; boundary=818305186e6894479422adc293157c862189c0dbd4c33fb3b2ef875535b4]]
Run: cannot run workloads: AddManifest: AddManifest: error send request: Post "https://:5532/api/v1/manifest": dial tcp :5532: connect: connection refused
Error: AddManifest: AddManifest: error send request: Post "https://:5532/api/v1/manifest": dial tcp :5532: connect: connection refused
Usage:
  aconcli run [manifest]... [flags]

Flags:
  -A, --all                load/start all images in the current ACON image repo
  -a, --auto               load dependencies automatically
  -c, --connect string     connect target url
      --env strings        set environment variables inside new containers
  -f, --file string        path to the executable for launching ACON TD/VM (default "/home/sgx/maliang/ACON/test/acon-startvm")
      --finalize           finalize the process of loading images to ACON TD/VM (default true)
  -h, --help               help for run
  -i, --interactive        run ACON TD/VM in foreground (usually for debugging)
  -l, --loadonly strings   load (but do not start) the specified images
  -n, --new                start a new ACON TD/VM
  -p, --proxy string       http proxy for ACON VM
  -t, --timetolive int     shut down the TD/VM after being idle for specified number of seconds (default 60)
  -u, --user string        user ID for OpenID authentication

Global Flags:
  -C, --directory string   change working directory before performing any operations
```

Reply:
@billionairiam  I think it was due to the platform you ran this test on was modified. When I enter the debug mode, I saw some problem with `qemu-kvm` :

`qemu-kvm: -accel kvm: KVM_TDX_CAPABILITIES failed: Inappropriate ioctl for device`